### PR TITLE
Sort versions smartly everywhere

### LINKED ISSUE #5151: Sort versions smartly everywhere
We currently have code that is smart about version sorting in Python:

https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/projects/templatetags/projects_tags.py#L13

We should make sure that we're using this everywhere that we display versions to a user. 

Refs https://github.com/rtfd/readthedocs.org/pull/5037.

### PR DESCRIPTION
**Pull request by dojutsu-user**
Closes #5151 

It wasn't possible to use the `sort_version_compare` only on the template level. So, I have also used it in view and forms level.

### CODE REVIEW

File: readthedocs/templates/core/version_list.html [hunk 1]
```diff
@@ -1,6 +1,8 @@
+{% load projects_tags %}
+
 <div class="module-list-wrapper">
```
Comment (dojutsu-user) [inline readthedocs/templates/core/version_list.html]: I didn't find any use case for this file.
Can it be deleted?

Comment (ericholscher) [reply readthedocs/templates/core/version_list.html]: Hrm, I think so!

**Comment by dojutsu-user** [comment]
@ericholscher 
Deleted the unused file.

**Comment by ericholscher** [comment]
💯 Thanks @dojutsu-user!

**Pull request merged**